webui/src/lib/api/index.js

```diff
@@ -118,6 +118,36 @@ export class MergeError extends Error {
 // actual actions:
 class Auth {
 
+    async updatePasswordByToken(token, newPassword) {
+        const response = await fetch(`${API_ENDPOINT}/auth/password`, {
+            headers: new Headers(defaultAPIHeaders),
+            method: 'POST',
+            body: json({token: token, newPassword: newPassword})
+        });
+
+        if (response.status === 401) {
+            throw new AuthorizationError('user unauthorized');
+        }
+        if (response.status !== 201) {
+            throw new Error('unknown authentication error');
+        }
+    }
+
+    async passwordForgot(email) {
+        const response = await fetch(`${API_ENDPOINT}/auth/password/forgot`, {
```

Comment:
wrote it also in the serve side pr - I think this flow requires captcha - it something that will require a human token as part of an unauthenticated request.